### DISCUSSION TITLE
Status-Pushbullet

### DISCUSSION THREAD

**Discussion by PascalSI** [taligentx/dscKeybusInterface#26] [dscKeybusInterface Discussion]
Hi, i tried to compile the Status_Pushbullet code, but it produces errors when compiling. Could you explain what could be the reason?

Here are all the errors:

Status_Pushbullet:61: error: 'class dscKeybusInterface' has no member named 'powerChanged'

     if (dsc.powerChanged) {
             ^
Status_Pushbullet:62: error: 'class dscKeybusInterface' has no member named 'powerChanged'
       dsc.powerChanged = false;  // Resets the battery trouble status flag
           ^
Status_Pushbullet:68: error: 'dscPartitions' was not declared in this scope
     for (byte partitionIndex = 0; partitionIndex < dscPartitions; partitionIndex++) {
                                                    ^
Status_Pushbullet:70: error: 'class dscKeybusInterface' has no member named 'alarmChanged'
       if (dsc.alarmChanged[partitionIndex]) {
               ^
Status_Pushbullet:71: error: 'class dscKeybusInterface' has no member named 'alarmChanged'
         dsc.alarmChanged[partitionIndex] = false;  // Resets the partition alarm status flag
             ^
Status_Pushbullet:78: error: 'class dscKeybusInterface' has no member named 'alarm'
         if (dsc.alarm[partitionIndex]) sendPush(pushMessage);
                 ^
Status_Pushbullet:82: error: 'class dscKeybusInterface' has no member named 'fireChanged'
       if (dsc.fireChanged[partitionIndex]) {
               ^
Status_Pushbullet:83: error: 'class dscKeybusInterface' has no member named 'fireChanged'
         dsc.fireChanged[partitionIndex] = false;  // Resets the fire status flag

**Comment by taligentx** [comment]
Hi @PascalSI, it looks like you're using the newer version of the sketch (with support for multiple partitions) with the older version of the library (supporting 1 partition only).

I've just released version 0.3, let's see if there are any issues after updating the library.

**Comment by PascalSI** [comment]
Yes, the compilation passed without errors, today I'll try with the security panel and push the results.
This version will work with PC1616 if there is only one partition on it?

**Comment by PascalSI** [comment]
Hi @taligentx , i somehow did not work, there are no errors when compiling errors, but when connected to the security panel, in addition to the message "Security system initializing", there is nothing.
I'll try again.

**Comment by taligentx** [comment]
If you're receiving that initialization message, the PushBullet side seems to be working.  I recommend checking to see if you are receiving data from the panel - load the `esp8266/KeyBusReader` example sketch and verify that the panel data is showing up and without CRC errors or invalid commands like `0x00`.  If that works, load the `esp8266/Status` sketch and check that you get status messages like Exit delay in progress, Armed stay/away, etc.  That should help narrow down the problem.

It should work fine with the PC1616 but if you see data coming into KeybusReader but nothing from the Status sketch, please post logs of KeybusReader while you are performing different actions (arm, disarm, etc) and I'll see if there's a decoding issue.

**Comment by PascalSI** [comment]
Ok, today or tomorrow I'll try what you advise and write the results.

**Comment by PascalSI** [comment]
I tried what you advised, but it did not help... Nothing is displayed in the port monitor, i do not know which way to look anymore, the divider of the preamble was assembled normally, i could not make mistakes in them, the ratings correspond to 15k and 10k, i have PC-1616 version 4.5.
I use the ESP8266 WEMOS D1

**Comment by PascalSI** [comment]
When the Status-Pushbullet firmware is started in the port monitor, it shows only the connection to the WiFi network:
I also noticed if i use D1 and D2 then the message in the port monitor is not complete, just this:
```
WiFi connected: 192.168.0.242
In
```

And if I use D3 and D4 then the message in the monitor of the port is different:
```
WiFi connected: 192.168.0.242
Initialization push notification sent successfully.
DSC Keybus Interface is online.

```

**Comment by PascalSI** [comment]
As I understand it, most likely the case in the bundle that connects PC-1616 and ESP8266 ...
In the port monitor, I can see that Esp8266 is working, there are messages about the initialization of programs such as "DSC Keybus Interface is online.", "Initialization push notification sent successfully.","Security system initializing" ... 
But there is no data from the panel.

**Comment by taligentx** [comment]
The `Status-PushBullet` sketch will not show any output to serial after it is connected to wifi.  For now I'd like to ignore the PushBullet example and only use the `KeybusReader` sketch for testing.  

If you can, take a photo of the PC1616 and the wiring to the esp8266 - let's see if there's a problem there.

**Comment by PascalSI** [comment]
Now I can take pictures with PC-1616, here is a board with resistances:
![2 1](https://user-images.githubusercontent.com/18159942/41588035-514b4336-73b9-11e8-9f72-69128aadf39d.JPG)
![6 1](https://user-images.githubusercontent.com/18159942/41588049-5a8fc6ec-73b9-11e8-88e5-52a187fb5f92.JPG)
![5 1](https://user-images.githubusercontent.com/18159942/41588056-5f6fb988-73b9-11e8-83ef-be819ef4eb8d.JPG)
It seems to me that everything is correct, the resistance is measured, they meet the nominal values +/- 5%...

**Comment by taligentx** [comment]
Thanks for the photos, if the labels are accurate I don't see an obvious problem.  If KeybusReader is not showing any data, it seems that there's either something like a broken wire, a short between connections, or problem with the esp8266 GPIO pins.

You can also try disconnecting all other keypads/modules from the panel and only have the Wemos connected to make sure there isn't some other interaction.

**Comment by taligentx** [comment]
@PascalSI Are you still having an issue with the wiring?

**Comment by taligentx** [comment]
Feel free to re-open this issue if you are continuing to troubleshoot the wiring.

**Comment by PascalSI** [comment]
Hi, I ordered a new module Esp, I'm waiting for it when I arrive. We do not sell them here.

**Comment by taligentx** [comment]
Okay, if there are any issues with the new module you can re-open this issue or create a new one.  For testing, you can try soldering the resistors and wires directly to each other instead of using a PCB or perfboard.  This may help narrow down the problem.
